common/service/dynamicconfig/constants.go

```diff
@@ -31,31 +31,97 @@ func (k Key) String() string {
 }
 
 const (
-	_matchingRoot               = "matching."
-	_matchingDomainTaskListRoot = _matchingRoot + "domain." + "taskList."
-	_historyRoot                = "history."
+	_matchingRoot = "matching."
+	_historyRoot  = "history."
+	_frontendRoot = "frontend."
 )
 
+// These are keys name dynamic config source use.
 var keys = []string{
 	"unknownKey",
-	_matchingDomainTaskListRoot + "minTaskThrottlingBurstSize",
-	_matchingDomainTaskListRoot + "maxTaskBatchSize",
-	_matchingDomainTaskListRoot + "longPollExpirationInterval",
-	_matchingDomainTaskListRoot + "enableSyncMatch",
-	_matchingDomainTaskListRoot + "updateAckInterval",
-	_matchingDomainTaskListRoot + "idleTasklistCheckInterval",
+	"testGetPropertyKey",
```

Comment:
Instead of having keys as a flat array, can we make it a map of service to list of configs?  
I think it will be much cleaner if we separate them out per service.  
Another thing is I don't like we rely on the ordering of how keys are defined.  Can we convert them into a map too, otherwise it will be very error prone.  
Basically the keys definition would look something like:
map[serviceIdx]map[int]string

Reply:
changed